cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -1005,6 +1035,8 @@ Status NumPyConverter::ConvertObjectsInfer() {
     } else if (PyDate_CheckExact(obj)) {
       // We could choose Date32 or Date64
       return ConvertDates<Date32Type>();
+    } else if (PyDateTime_CheckExact(obj)) {
```

Comment:
I opened https://issues.apache.org/jira/browse/ARROW-2130 about dealing with pandas.Timestamp which will inevitably pop up

Reply:
I'll take care of that in a follow-up PR.